tests/app/main/test_services.py

```diff
@@ -303,15 +330,48 @@ def test_should_view_disabled_service_with_removed_message(
             res.get_data(as_text=True)
         )
 
-        self._post_remove_service(service_should_be_modifiable=False)
+    def test_should_not_view_other_suppliers_services(
+            self, data_api_client, supplier_service_editing_fw_params
+    ):
+        framework_slug, framework_name, framework_editable_services = supplier_service_editing_fw_params
+        self.login()
+        self._setup_service(
+            data_api_client, framework_slug, framework_name, service_status='published', service_belongs_to_user=False)
 
+        res = self.client.get('/suppliers/services/123')
+        if not framework_editable_services:
+            assert_equal(res.status_code, 404)
+            return
+
+        assert_equal(res.status_code, 404)
+
+    def test_should_redirect_to_login_if_not_logged_in(self, data_api_client):
+        res = self.client.get("/suppliers/services/123")
+        assert_equal(res.status_code, 302)
+        assert_equal(res.location,
+                     'http://localhost/login?next=%2Fsuppliers%2Fservices%2F123')
+
+
+@mock.patch('app.main.views.services.data_api_client')
+class TestSupplierRemoveServiceEditInterplay(_BaseTestSupplierEditRemoveService):
+    """
+        These tests actually test the *interplay* between the remove view and its subsequent (redirected)
+        views through using `follow_redirects` to perform both a POST to the remove view and a subsequent GET to the
+        following view. Chief thing we're asserting is the flash message throw/catch.
+    """
     def test_should_view_confirmation_message_if_first_remove_service_button_clicked(
-            self, data_api_client, fixture_framework_slug_and_name
+            self, data_api_client, supplier_service_editing_fw_params
     ):
+        framework_slug, framework_name, framework_editable_services = supplier_service_editing_fw_params
         self.login()
-        self._get_service(data_api_client, fixture_framework_slug_and_name, service_status='published')
+        self._setup_service(data_api_client, framework_slug, framework_name, service_status='published')
 
+        # NOTE two http requests performed here
         res = self.client.post('/suppliers/services/123/remove', follow_redirects=True)
+        if not framework_editable_services:
+            assert_equal(res.status_code, 404)
+            assert_is(data_api_client.update_service_status.called, False)
```

Comment:
The nose-removal patch sits on top of this one (*I think*) so the nose-removal happens after this - I guess I just wanted to keep it symmetrical with the rest of the code as shown?